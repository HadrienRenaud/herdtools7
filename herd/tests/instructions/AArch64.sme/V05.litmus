AArch64 V05
{
  uint32_t x[4];
  0:X0=x;
}
P0 ;
SMSTART                      ;
INDEX Z0.S,#1,#1             ;
INDEX Z1.S,#2,#2             ;
INDEX Z2.S,#3,#3             ;
INDEX Z3.S,#4,#4             ;
PTRUE P0.S                   ;
MOV W12,WZR                  ;
MOVA ZA0H.S[W12,0],P0/M,Z0.S ;
MOVA ZA0H.S[W12,1],P0/M,Z1.S ;
MOVA ZA0H.S[W12,2],P0/M,Z2.S ;
MOVA ZA0H.S[W12,3],P0/M,Z3.S ;
PTRUE P1.S,VL4               ;
ST1W {ZA0V.S[W12,0]},P1,[X0] ;
SMSTOP                       ;

forall x={1,2,3,4}
