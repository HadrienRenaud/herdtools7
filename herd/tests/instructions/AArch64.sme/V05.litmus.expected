Test V05 Required
States 1
x={1,2,3,4};
Ok
Witnesses
Positive: 1 Negative: 0
Flag Scalable-Matrix-Extensions-is-work-in-progress
Condition forall (x={1,2,3,4})
Observation V05 Always 1 0
Hash=0cf19d9833300d409985c4f4c607de19

